java-compiler-testing/src/main/java/io/github/ascopes/jct/workspaces/impl/TempDirectory.java

```diff
@@ -44,64 +45,40 @@
  * @see RamDirectory
  * @since 0.0.1
  */
-@API(since = "0.0.1", status = Status.EXPERIMENTAL)
-public final class TempDirectory extends AbstractTestDirectory<TempDirectory> {
+@API(since = "0.0.1", status = Status.INTERNAL)
+public final class TempDirectory extends TestDirectory {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(TempDirectory.class);
 
-  private TempDirectory(
-      String name,
-      Path rootDirectory,
-      String separator,
-      boolean closeOnGc,
-      Closeable closeHook
-  ) {
-    super(name, rootDirectory, separator, closeOnGc, closeHook);
+  private final Path rootDirectory;
+
+  private TempDirectory(String name, Path rootDirectory, String separator) {
+    super(name, rootDirectory, separator);
+    this.rootDirectory = rootDirectory;
+  }
+
+  @Override
+  public void close() throws IOException {
+    LOGGER.debug(
+        "Deleting temporary directory ({} @ {})",
+        rootDirectory.toUri(),
+        rootDirectory.getFileSystem()
+    );
+    RecursiveDeleter.deleteAll(rootDirectory);
```

Comment:
- [ ] If we do not use the deleter elsewhere, we may be able to inline this call and remove that class.